cross-compiling/advanced.md

```diff
@@ -0,0 +1,145 @@
+rm -rf tinyxml2
+```
+
+Note that if your newly-added dependency is not being statically linked to the ROS 2 packages, you will have to install it on your target system as well.
+
+### Applying patches
+
+It may happen that the ROS 2 package that you downloaded is not suitable for being cross-compiled as it is.
+If the cross-compilation fails, you may want to address the failure by applying a patch to the failing ROS 2 package.
+
+You can either fork the ROS 2 package and apply the patch to your version or you can keep using the standard version and let the cross-compilation framework to apply the patch.
+
+In the second case, you will need to add to `ignore_pkgs_scripts/<TARGET_ARCHITECTURE>_ignore.sh` something like
+```
+patch -p0 < $THIS_DIR/<TARGET_ARCHITECTURE>_my_changes.patch
+```
+
+Where `ignore_pkgs_scripts/<TARGET_ARCHITECTURE>_my_changes.patch` contains the patch that you want to apply.
+
+## Debugging of cross-compilation issues
+
+When you add new ROS 2 packages to your cross-compilation workspace, it's common that you will face compilation errors.
+The terminal will show WARN level logs.
+You can find all logs and all build commands in the directory `ros2_cc_ws/log/latest_build/<ROS2_PACKAGE>`.
+
+To speed-up debugging, it is recommended to manually invoke the build commands.
+The `--debug` option will bring you inside the cross-compilation Docker environment
+
+```
+bash cc_workspace.sh ~/ros2_cc_ws --debug
+```
+
+To start a build, **from within the Docker environment**
+```
+bash /root/compilation_scripts/cross_compile.sh
+```
+
+You can modify the cross_compile script `/root/compilation_scripts/cross_compile.sh` **from within the Docker environment** to customize your debugging.
+Common changes are the following:
+
+ - Comment the line that clears the workspace every time to speed up builds `rm -rf build install log`
+ - Build only the package that is failing, adding one of the following to the `colcon build` arguments: `--packages-up-to MY_PACKAGE` or `--packages-select MY_PACKAGE`.
+ - Change the colcon log verbosity adding `--log-level debug`
+
+## Cross-compile individual ROS 2 packages
+
+If for any reason you don't want to cross-compile a whole ROS 2 workspace made of all the ROS 2 core packages, but you only want to cross-compile a specific individual package, you can do that by first cross-compiling the standard ROS 2 SDK, as described in the [README](README.md), and then you can copy the cross-compiled libraries into the sysroot of your target architecture.
+
+```
+cp -R ~/ros2_cc_ws/install/cmake sysroots/<TARGET_ARCHITECTURE>/usr/cmake
+cp -R ~/ros2_cc_ws/install/share/* sysroots/<TARGET_ARCHITECTURE>/usr/share/
+cp -R ~/ros2_cc_ws/install/lib/* sysroots/<TARGET_ARCHITECTURE>/usr/lib/
+cp -R ~/ros2_cc_ws/install/include/* sysroots/<TARGET_ARCHITECTURE>/usr/include/
+```
+
+**NOTE:** calling `bash get_sysroot.sh` will erase the current sysroot and generate a new one from scratches, thus losing the copied ROS 2 SDK.
```

Comment:
We should do same as in `get_ros2_sources.sh`, i.e. warn that the directory already exsist. The user should remove it first manually

Reply:
Yes, agreed